providers/fips/fipsprov.c

```diff
@@ -274,6 +274,12 @@ static const OSSL_ALGORITHM fips_ciphers[] = {
     { NULL, NULL, NULL }
 };
 
+static const OSSL_ALGORITHM fips_macs[] = {
+    { "GMAC", "default=yes", gmac_functions },
+    { "HMAC", "default=yes", hmac_functions },
```

Comment:
NIST has standards for CMAC and KMAC.  The two KMACs are up above in the digests, I thought it was plumbed in via MAC.  @slontis you did this didn't you?

CMAC is used as part of other things (memory fails me as to what) and is more common than GMAC.  KMAC isn't common yet.

Even if extra algorithms are included in the FIPS provider, they don't need to be part of the validation.  Non-approved algorithms are allowed and get documented as such.  I kind of suspect that the lab isn't going to object to any extras, they'll have to do validations for them as well in due course, so no effort gets saved.  I've been meaning to ask about this.

Reply:
I'm surprised you didn't comment on the utterly faulty property! :wink:

Reply:
Regarding the use as MD or as MAC, I'm honestly utterly confused.  What @slontis has implemented is a keyed hash, and sure, that's essentially what a MAC is...  but how do we want to present that to the users, as a `EVP_MD` or as a `EVP_MAC`?  We're currently doing both.

Reply:
Regarding the choice of MACs to be included in the FIPS provider module, I relied on the choices made in https://www.openssl.org/docs/OpenSSL300Design.html, see Appendix 3.

Reply:
Meanwhile, I fixed the property strings

Reply:
Yet, KMAC is included.  I still think that there is some merit in including them in the FIPS module even if they aren't immediately validated.  It would make validating them easier and we might get lucky with the lab :)

@slontis, thoughts about the MD vs MAC for KMAC?

Reply:
Well its name kind of gives it away :) It should be a MAC.

Reply:
So why do we have it implemented as a hash?

Reply:
Specifically, these two lines confused the hell out of me:

https://github.com/openssl/openssl/blob/697b0c5185f2b379cf23330fddc5f8b2d691db17/providers/common/digests/sha3_prov.c#L287-L288

Reply:
I don't think they should be hashes, just MACs.  That seems to have slipped through.

Reply:
> Specifically, these two lines confused the hell out of me:

Yep, it was done for a reason..

Reply:
That makes sense (and helps restore my sanity, thank you :wink:)

Reply:
And oh, I did move the KMAC here now...

Reply:
Ahh yeah, I must of got confused when I did it...
KMAC/SHA3 use the same KECCAK functionality.

Reply:
Ah, I see why it's there...  `crypto/evp/m_sha3.c` implements KMAC as a EVP_MD, because....  well, because what, really?  Was it so it could be used for PRF purposes?

Reply:
The single step KDF uses it..

Reply:
Aaaaanyway...  I'll have to adjust some things around KMAC.

Reply:
Done